adapter/service/chargepoint/service.go

```diff
@@ -456,16 +470,31 @@ func (s *service) SupportsAdjustingPhaseModes() bool {
 	return err == nil
 }
 
-// adjustableCurrentController returns the AdjustableCurrentController, if supported.
-func (s *service) adjustableCurrentController() (AdjustableCurrentController, error) {
+// adjustableMaxCurrentController returns the AdjustableMaxCurrentController, if supported.
+func (s *service) adjustableMaxCurrentController() (AdjustableMaxCurrentController, error) {
 	_, ok := s.Specification().PropertyInteger(PropertySupportedMaxCurrent)
 	if !ok {
-		return nil, fmt.Errorf("%s: adjusting current is not supported", s.Name())
+		return nil, fmt.Errorf("%s: adjusting max current is not supported", s.Name())
+	}
+
+	controller, ok := s.controller.(AdjustableMaxCurrentController)
+	if !ok {
+		return nil, fmt.Errorf("%s: adjusting max current is not supported", s.Name())
+	}
+
+	return controller, nil
+}
+
+// adjustableOfferedCurrentController returns the AdjustableOfferedCurrentController, if supported.
+func (s *service) adjustableOfferedCurrentController() (AdjustableOfferedCurrentController, error) {
+	_, ok := s.Specification().PropertyInteger(PropertyOfferedCurrent)
```

Comment:
Nope, this is incorrect. PropertyOfferedCurrent is a property of "evt.current_session.report" interface, not a service specification.

What you should check here, is if there is sup_max_current property defining an upper boundary for setting offered current.

Reply:
The logic dictates we have this requirement for both of the controllers.